api/helpers/uploadedFileErrorChecks.js

```diff
@@ -291,32 +284,15 @@ function checkTypeSizeAndDuplication(req, file, cb) {
     }
 
     if (errors.length > 0) {
-        addErrorsToSession(req, file.originalname, errors);
+        addErrorsToPage(req, file.originalname, errors);
         preventFileUpload();
     } else {
         allowFileUplaod();
     }
 }
 
-function addErrorsToSession(req, fileName, errors) {
-    const fileNamesWithErrors = req.session.eApp.uploadMessages.errors.map(
-        (error) => error.hasOwnProperty('filename') && error.filename
-    );
-    if (fileNamesWithErrors.includes(fileName)) {
-        fileNamesWithErrors.forEach((fileNameFromSession, idx) => {
-            if (fileNameFromSession === fileName) {
-                req.session.eApp.uploadMessages.errors[idx].errors = [
-                    ...req.session.eApp.uploadMessages.errors[idx].errors,
-                    ...errors,
-                ];
-            }
-        });
-    } else {
-        req.session.eApp.uploadMessages.errors.push({
-            filename: fileName,
-            errors,
-        });
-    }
+function addErrorsToPage(req, filename, errors) {
+    req.flash('errors', [{ filename, errors }]);
```

Comment:
is this necessary? 

to add an error to a page, you would just do req.flash('errors', ['myErrors']). I think also you've used a mixture of addErrorsToPage and req.flash in some places, which may be confusing things. Stick to one - I would just go with req.flash

Reply:
Good point